### ISSUE TITLE
ROOT messed up clean build of icerec by creating linking issues (Trac #796)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#790]
Downloaded latest trunk to my local machine (Mac Mavericks).
Had problems with running dataio-shovel, dataio-pyshovel and steamshovel:

```text
rstrom@bubbles:~/Icecode/meta-projects/icerec/build_release>dataio-pyshovel
Traceback (most recent call last):
  File "/Users/rstrom/Icecode/meta-projects/icerec/build_release/bin/dataio-pyshovel", line 7, in <module>
    from icecube import icetray
  File "/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.py", line 2, in <module>
    load_pybindings(__name__, __path__)
  File "/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.py", line 58, in load_pybindings
    m = imp.load_dynamic(name, path[0] + ".so")
ImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv
  Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
  Expected in: /Users/rstrom/Icecode/meta-projects/icerec/build_release/lib/libicetray.dylib
 in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
```
I tried to fix this with upgrading boost (from homebrew) and I realized that boost-python was missing from homebrew, although it seemed like cmake picked it up in the configuration. I ended up reinstalling boost and boost-python completely but I still had problems.

The problems propagated to the limit where nothing worked, I could not even load icetray or dataclasses.
I got the following error message:

```text
In [1]: from icecube import dataclasses
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-7dd057f6475f> in <module>()
----> 1 from icecube import dataclasses

/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/dataclasses/__init__.py in <module>()
      1 from icecube.load_pybindings import load_pybindings
----> 2 import icecube.icetray # be nice and pull in our dependencies
      3 
      4 # XXX Work around a stupid bug with certain combinations of Boost, compiler, and
      5 # RTLD where pulling in libdataclasses as a dependency of the pybindings will

/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.pyc in <module>()
      1 from icecube.load_pybindings import load_pybindings
----> 2 load_pybindings(__name__, __path__)
      3 
      4 from icecube.icetray.test_module import I3TestModuleFactory
      5 from icecube.icetray.traysegment import traysegment, traysegment_inherit, module_altconfig

/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.pyc in load_pybindings(name, path)
     56     import imp, sys
     57     thismod = sys.modules[name]
---> 58     m = imp.load_dynamic(name, path[0] + ".so")
     59     sys.modules[name] = thismod # Some python versions overwrite the Python
     60                                 # module entry with the C++ one. We don't want

ImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv
  Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
  Expected in: /Users/rstrom/Icecode/meta-projects/icerec_V04-09-00/build_release/lib/libicetray.dylib
 in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
```
JVS suggested I build without ROOT. Now working.

Conclusion: Don't use root!

Migrated from https://code.icecube.wisc.edu/ticket/796
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T23:12:56", 
    "description": "Downloaded latest trunk to my local machine (Mac Mavericks).\nHad problems with running dataio-shovel, dataio-pyshovel and steamshovel:\n{{{\nrstrom@bubbles:~/Icecode/meta-projects/icerec/build_release>dataio-pyshovel\nTraceback (most recent call last):\n  File \"/Users/rstrom/Icecode/meta-projects/icerec/build_release/bin/dataio-pyshovel\", line 7, in <module>\n    from icecube import icetray\n  File \"/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.py\", line 2, in <module>\n    load_pybindings(__name__, __path__)\n  File \"/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.py\", line 58, in load_pybindings\n    m = imp.load_dynamic(name, path[0] + \".so\")\nImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv\n  Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so\n  Expected in: /Users/rstrom/Icecode/meta-projects/icerec/build_release/lib/libicetray.dylib\n in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so\n}}}\nI tried to fix this with upgrading boost (from homebrew) and I realized that boost-python was missing from homebrew, although it seemed like cmake picked it up in the configuration. I ended up reinstalling boost and boost-python completely but I still had problems.\n\nThe problems propagated to the limit where nothing worked, I could not even load icetray or dataclasses.\nI got the following error message:\n{{{\nIn [1]: from icecube import dataclasses\n---------------------------------------------------------------------------\nImportError                               Traceback (most recent call last)\n<ipython-input-1-7dd057f6475f> in <module>()\n----> 1 from icecube import dataclasses\n\n/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/dataclasses/__init__.py in <module>()\n      1 from icecube.load_pybindings import load_pybindings\n----> 2 import icecube.icetray # be nice and pull in our dependencies\n      3 \n      4 # XXX Work around a stupid bug with certain combinations of Boost, compiler, and\n      5 # RTLD where pulling in libdataclasses as a dependency of the pybindings will\n\n/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.pyc in <module>()\n      1 from icecube.load_pybindings import load_pybindings\n----> 2 load_pybindings(__name__, __path__)\n      3 \n      4 from icecube.icetray.test_module import I3TestModuleFactory\n      5 from icecube.icetray.traysegment import traysegment, traysegment_inherit, module_altconfig\n\n/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.pyc in load_pybindings(name, path)\n     56     import imp, sys\n     57     thismod = sys.modules[name]\n---> 58     m = imp.load_dynamic(name, path[0] + \".so\")\n     59     sys.modules[name] = thismod # Some python versions overwrite the Python\n     60                                 # module entry with the C++ one. We don't want\n\nImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv\n  Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so\n  Expected in: /Users/rstrom/Icecode/meta-projects/icerec_V04-09-00/build_release/lib/libicetray.dylib\n in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so\n}}}\nJVS suggested I build without ROOT. Now working.\n\nConclusion: Don't use root!", 
    "reporter": "rstrom", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547248376057503", 
    "component": "combo reconstruction", 
    "summary": "ROOT messed up clean build of icerec by creating linking issues", 
    "priority": "normal", 
    "keywords": "root, icerec, lib", 
    "time": "2014-11-05T20:36:54", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20141112T17:09:06: nega changed reporter from "icecube" to "rstrom"

**Comment by icecube-trac** [comment]
Trac update at 20150211T21:27:23: 
* nega changed status from "new" to "assigned"
* nega changed owner from "" to "nega"

**Comment by icecube-trac** [comment]
Trac update at 20150212T23:06:45: 
* nega changed resolution from "" to "fixed"
* nega changed status from "assigned" to "closed"
* nega commented:

> In changeset 2071:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2071"
> update docs w/ rstrom's mystery ROOT fail
> 
> closes #790
> ```

**Comment by icecube-trac** [comment]
Trac update at 20150212T23:07:54: nega changed description from:

> Downloaded latest trunk to my local machine (Mac Mavericks).
> Had problems with running dataio-shovel, dataio-pyshovel and steamshovel:
> 
> rstrom@bubbles:~/Icecode/meta-projects/icerec/build_release>dataio-pyshovel
> Traceback (most recent call last):
>   File "/Users/rstrom/Icecode/meta-projects/icerec/build_release/bin/dataio-pyshovel", line 7, in <module>
>     from icecube import icetray
>   File "/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.py", line 2, in <module>
>     load_pybindings(__name__, __path__)
>   File "/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.py", line 58, in load_pybindings
>     m = imp.load_dynamic(name, path[0] + ".so")
> ImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv
>   Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
>   Expected in: /Users/rstrom/Icecode/meta-projects/icerec/build_release/lib/libicetray.dylib
>  in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
> 
> I tried to fix this with upgrading boost (from homebrew) and I realized that boost-python was missing from homebrew, although it seemed like cmake picked it up in the configuration. I ended up reinstalling boost and boost-python completely but I still had problems.
> 
> The problems propagated to the limit where nothing worked, I could not even load icetray or dataclasses.
> I got the following error message:
> 
> In [1]: from icecube import dataclasses
> ---------------------------------------------------------------------------
> ImportError                               Traceback (most recent call last)
> <ipython-input-1-7dd057f6475f> in <module>()
> ----> 1 from icecube import dataclasses
> 
> /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/dataclasses/__init__.py in <module>()
>       1 from icecube.load_pybindings import load_pybindings
> ----> 2 import icecube.icetray # be nice and pull in our dependencies
>       3 
>       4 # XXX Work around a stupid bug with certain combinations of Boost, compiler, and
>       5 # RTLD where pulling in libdataclasses as a dependency of the pybindings will
> 
> /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.pyc in <module>()
>       1 from icecube.load_pybindings import load_pybindings
> ----> 2 load_pybindings(__name__, __path__)
>       3 
>       4 from icecube.icetray.test_module import I3TestModuleFactory
>       5 from icecube.icetray.traysegment import traysegment, traysegment_inherit, module_altconfig
> 
> /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.pyc in load_pybindings(name, path)
>      56     import imp, sys
>      57     thismod = sys.modules[name]
> ---> 58     m = imp.load_dynamic(name, path[0] + ".so")
>      59     sys.modules[name] = thismod # Some python versions overwrite the Python
>      60                                 # module entry with the C++ one. We don't want
> 
> ImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv
>   Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
>   Expected in: /Users/rstrom/Icecode/meta-projects/icerec_V04-09-00/build_release/lib/libicetray.dylib
>  in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
> 
> JVS suggested I build without ROOT. Now working.
> 
> Conclusion: Don't use root!

to:

> Downloaded latest trunk to my local machine (Mac Mavericks).
> Had problems with running dataio-shovel, dataio-pyshovel and steamshovel:
> {{{
> rstrom@bubbles:~/Icecode/meta-projects/icerec/build_release>dataio-pyshovel
> Traceback (most recent call last):
>   File "/Users/rstrom/Icecode/meta-projects/icerec/build_release/bin/dataio-pyshovel", line 7, in <module>
>     from icecube import icetray
>   File "/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.py", line 2, in <module>
>     load_pybindings(__name__, __path__)
>   File "/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.py", line 58, in load_pybindings
>     m = imp.load_dynamic(name, path[0] + ".so")
> ImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv
>   Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
>   Expected in: /Users/rstrom/Icecode/meta-projects/icerec/build_release/lib/libicetray.dylib
>  in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
> }}}
> I tried to fix this with upgrading boost (from homebrew) and I realized that boost-python was missing from homebrew, although it seemed like cmake picked it up in the configuration. I ended up reinstalling boost and boost-python completely but I still had problems.
> 
> The problems propagated to the limit where nothing worked, I could not even load icetray or dataclasses.
> I got the following error message:
> {{{
> In [1]: from icecube import dataclasses
> ---------------------------------------------------------------------------
> ImportError                               Traceback (most recent call last)
> <ipython-input-1-7dd057f6475f> in <module>()
> ----> 1 from icecube import dataclasses
> 
> /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/dataclasses/__init__.py in <module>()
>       1 from icecube.load_pybindings import load_pybindings
> ----> 2 import icecube.icetray # be nice and pull in our dependencies
>       3 
>       4 # XXX Work around a stupid bug with certain combinations of Boost, compiler, and
>       5 # RTLD where pulling in libdataclasses as a dependency of the pybindings will
> 
> /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray/__init__.pyc in <module>()
>       1 from icecube.load_pybindings import load_pybindings
> ----> 2 load_pybindings(__name__, __path__)
>       3 
>       4 from icecube.icetray.test_module import I3TestModuleFactory
>       5 from icecube.icetray.traysegment import traysegment, traysegment_inherit, module_altconfig
> 
> /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/load_pybindings.pyc in load_pybindings(name, path)
>      56     import imp, sys
>      57     thismod = sys.modules[name]
> ---> 58     m = imp.load_dynamic(name, path[0] + ".so")
>      59     sys.modules[name] = thismod # Some python versions overwrite the Python
>      60                                 # module entry with the C++ one. We don't want
> 
> ImportError: dlopen(/Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so, 2): Symbol not found: __Z21register_std_cont_podv
>   Referenced from: /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
>   Expected in: /Users/rstrom/Icecode/meta-projects/icerec_V04-09-00/build_release/lib/libicetray.dylib
>  in /Users/rstrom/Icecode/meta-projects/icerec_trunk/build_release/lib/icecube/icetray.so
> }}}
> JVS suggested I build without ROOT. Now working.
> 
> Conclusion: Don't use root!

**Comment by icecube-trac** [comment]
Trac update at 20150707T22:32:33: 
* olivas changed milestone from "IceRec V4" to ""
* olivas commented:

> Milestone IceRec V4 deleted

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:12:56: nwhitehorn commented:

> In changeset 169928:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="169928"
> update docs w/ rstrom's mystery ROOT fail
> 
> closes #790
> ```
